sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -390,6 +391,167 @@ abstract class UnixTimeExprMeta[A <: BinaryExpression with TimeZoneAwareExpressi
   }
 }
 
+trait GpuNumberToTimestampUnaryExpression extends GpuUnaryExpression {
+  
+  override def dataType: DataType = TimestampType
+  override def outputTypeOverride: DType = DType.TIMESTAMP_MICROSECONDS
+
+  /**
+   * Test whether if input * multiplier will cause Long-overflow. In Math.multiplyExact, 
+   * if there is an integer-overflow, then it will throw an ArithmeticException "long overflow"
+   */
+  def checkLongMultiplicationOverflow(input: ColumnVector, multiplier: Long): Unit = {
+    withResource(input.max()) { maxValue =>
+      if (maxValue.isValid) {
+        Math.multiplyExact(maxValue.getLong, multiplier)
+      }
+    }
+    withResource(input.min()) { minValue =>
+      if (minValue.isValid) {
+        Math.multiplyExact(minValue.getLong, multiplier)
+      }
+    }
+  }
+
+  protected val convertTo : GpuColumnVector => ColumnVector
+  
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    convertTo(input)
+  }
+}
+
+case class GpuSecondsToTimestamp(child: Expression) extends GpuNumberToTimestampUnaryExpression {
+
+  override def nullable: Boolean = child.dataType match {
+    case _: FloatType | _: DoubleType => true
+    case _ => child.nullable
+  }
+
+  protected lazy val convertTo: GpuColumnVector => ColumnVector = child.dataType match {
+    case LongType =>
+      (input: GpuColumnVector) => {
+        checkLongMultiplicationOverflow(input.getBase, DateTimeConstants.MICROS_PER_SECOND)
+        val mul = withResource(Scalar.fromLong(DateTimeConstants.MICROS_PER_SECOND)) { scalar =>
+          input.getBase.mul(scalar)
+        }
+        withResource(mul) { mul =>
+          mul.asTimestampMicroseconds()
+        }
+      }
+    case DoubleType | FloatType =>
+      (input: GpuColumnVector) => {
+        GpuCast.doCast(input.getBase, input.dataType, TimestampType, false, false, false)
+      }
+    case dt: DecimalType =>
+      (input: GpuColumnVector) => {
+        // SecondsToTimestamp supports decimals with a scale of 6 or less, which can be represented
+        // as microseconds. An exception will be thrown if the scale is more than 6.
+        val decimalTypeSupported = DType.create(DType.DTypeEnum.DECIMAL128, -6)
+        if (dt.scale > 6) {
```

Comment:
NIT:
We can put this rounding check in a separate method, just like `checkLongMultiplicationOverflow`.